mayastor/src/bdev/nexus/nexus_io.rs

```diff
@@ -576,114 +473,72 @@ impl NexusBio {
-                                "core {} thread {:?}, faulting child {}",
-                                Cores::current(),
-                                Mthread::current(),
-                                device,
-                            );
-
-                            // Pausing a nexus acts like entering a critical
-                            // section,
-                            // allowing only one retire request to run at a
-                            // time, which prevents
-                            // inconsistency in reading/updating nexus
-                            // configuration.
-                            nexus.pause().await.unwrap();
-                            nexus.set_failfast().await.unwrap();
-                            nexus.reconfigure(DrEvent::ChildFault).await;
-
-                            // Lookup child once more and finally remove it.
-                            match nexus.child_lookup(&device) {
-                                Some(child) => {
-                                    nexus
-                                        .persist(PersistOp::Update((
-                                            child.name.clone(),
-                                            child.state(),
-                                        )))
-                                        .await;
-                                    // TODO: an error can occur here if a
-                                    // separate task,
-                                    // e.g. grpc request is also deleting the
-                                    // child.
-                                    if let Err(err) = child.destroy().await {
-                                        error!(
-                                            "{}: destroying child {} failed {}",
-                                            nexus, child, err
-                                        );
-                                    }
-                                }
-                                None => {
-                                    warn!(
-                                        "{} no longer belongs to nexus {}, skipping child removal",
-                                        device, nexus
-                                    );
-                                }
-                            }
-
-                            nexus.clear_failfast().await.unwrap();
-                            nexus.resume().await.unwrap();
-
-                            if nexus.status() == NexusStatus::Faulted {
-                                error!(":{} has no children left... ", nexus);
-                            }
-                        }
-                    }
-                }
-            }
-            None => {
-                debug!("{} Nexus does not exist anymore", nexus);
+        if let Some(nexus) = nexus_lookup(&nexus) {
+            warn!(?nexus, ?device, "retiring child");
+
+            nexus.child_retire(device).await.unwrap();
```

Comment:
I'm getting this panic on a manual test of 3 nexuses with 3 replicas each running fio with the spdk ioengine, then terminating the Mayastor instance hosting all the replicas:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: PauseError { state: Pausing, name: "nexus-5b5b04ea-c1e3-11ea-bd82-a7d5cb04b390" }', mayastor/src/bdev/nexus/nexus_io.rs:539:46
```

Reply:
So, all replicas fail at the same time? or.. ?

Reply:
Essentially yes, as all 9 replicas are hosted in the same instance of Mayastor. The panic only happens with a release build though, I can't make it happen with a debug build.